CMakeLists.txt

```diff
@@ -75,6 +81,142 @@ target_compile_definitions(notcurses
     FORTIFY_SOURCE=2 _GNU_SOURCE SOURCE_DEFAULT
 )
 
+# libnotcurses++
+set(NCPP_SOURCES
+  src/libcpp/NotCurses.cc
+  src/libcpp/PanelReel.cc
+  src/libcpp/Plane.cc
+  src/libcpp/Root.cc
+  src/libcpp/Tablet.cc
+  src/libcpp/Visual.cc
+  )
+
+add_library(
+  notcurses++
+  SHARED
+  ${NCPP_SOURCES}
+  )
+
+add_library(
+  notcurses++-static
+  STATIC
+  ${NCPP_SOURCES}
+  )
+set_target_properties(
+  notcurses++-static PROPERTIES
+  OUTPUT_NAME notcurses++
+  )
+
+set_target_properties(
+  notcurses++ PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+  OUTPUT_NAME "notcurses++")
+
+set(NCPP_INCLUDE_DIRS
+  include
+    "${PROJECT_BINARY_DIR}/include"
+    "${TERMINFO_INCLUDE_DIRS}"
+  )
+
+target_include_directories(notcurses++
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_include_directories(notcurses++-static
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_link_libraries(notcurses++
+  PRIVATE
+  notcurses)
+
+set(NCPP_COMPILE_OPTIONS
+  -Wall
+  -Wextra
+  -W
+  -Wshadow
+  -Wformat
+  -Werror=format-security
+  -Wnull-dereference
+  -Wmisleading-indentation
+  -Wunused
+  -Wpedantic
+  -Wsuggest-override
+  -Wno-c99-extensions
+  -fno-strict-aliasing
+  -ffunction-sections
+  -funswitch-loops
+  -finline-limit=300
+  -fstack-protector
+  -fno-rtti
+  -fno-exceptions
+  -mpopcnt
+  -fpic
+  -pthread
```

Comment:
We should turn this on with the Pthreads package support, like I do in CMakeFiles.txt, so that it's portable (or some hopeful approximation thereof).

Reply:
Yeah, good idea.